From 39051951a2d27d035733635f0ba5bc070259bc73 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:16 -0400
Subject: [fs] btrfs: remove the unnecessary flush when preparing the pages

Message-id: <1407179109-13714-250-git-send-email-sandeen@redhat.com>
Patchwork-id: 87191
O-Subject: [RHEL7.1 PATCH 249/422] Btrfs: remove the unnecessary flush when preparing the pages
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Miao Xie <miaox@cn.fujitsu.com>

commit b88935bf9822cda58fd70dffe8e016d448757d40
Author: Miao Xie <miaox@cn.fujitsu.com>
Date:   Thu Mar 6 13:54:58 2014 +0800

    Btrfs: remove the unnecessary flush when preparing the pages

    Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/file.c b/fs/btrfs/file.c
index 72dcb37..7401031 100644
--- a/fs/btrfs/file.c
+++ b/fs/btrfs/file.c
@@ -1360,11 +1360,11 @@ lock_and_cleanup_extent_if_need(struct inode *inode, struct page **pages,
   struct btrfs_ordered_extent *ordered;
   lock_extent_bits(&BTRFS_I(inode)->io_tree,
      start_pos, last_pos, 0, cached_state);
-  ordered = btrfs_lookup_first_ordered_extent(inode, last_pos);
+  ordered = btrfs_lookup_ordered_range(inode, start_pos,
+           last_pos - start_pos + 1);
   if (ordered &&
       ordered->file_offset + ordered->len > start_pos &&
       ordered->file_offset <= last_pos) {
-   btrfs_put_ordered_extent(ordered);
    unlock_extent_cached(&BTRFS_I(inode)->io_tree,
           start_pos, last_pos,
           cached_state, GFP_NOFS);
@@ -1372,12 +1372,9 @@ lock_and_cleanup_extent_if_need(struct inode *inode, struct page **pages,
     unlock_page(pages[i]);
     page_cache_release(pages[i]);
    }
-   ret = btrfs_wait_ordered_range(inode, start_pos,
-      last_pos - start_pos + 1);
-   if (ret)
-    return ret;
-   else
-    return -EAGAIN;
+   btrfs_start_ordered_extent(inode, ordered, 1);
+   btrfs_put_ordered_extent(ordered);
+   return -EAGAIN;
   }
   if (ordered)
    btrfs_put_ordered_extent(ordered);
-- 
1.7.1